lib/Conversion/TorchToLinalg/TorchToLinalg.cpp

```diff
@@ -3594,6 +3594,72 @@ class ConvertAtenIndexSelectOp : public OpConversionPattern<AtenIndexSelectOp> {
 };
 } // namespace
 
+namespace {
+class ConvertAtenCumsumOp : public OpConversionPattern<AtenCumsumOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenCumsumOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+
+    int64_t dim;
+    if (!matchPattern(op.dim(), m_TorchConstantInt(&dim)))
+      return rewriter.notifyMatchFailure(op, "dim must be constant");
+
+    Location loc = op.getLoc();
+    MLIRContext *context = op->getContext();
+    Value input = adaptor.self();
+    auto inputType = input.getType().cast<RankedTensorType>();
+    int64_t inputRank = inputType.getRank();
+    Type elementType = inputType.getElementType();
+
+    Value zero = rewriter.create<arith::ConstantIndexOp>(loc, 0);
+    SmallVector<AffineExpr> indexExprs;
+    for (int i = 0; i < inputRank; i++) {
+      if (i == dim) {
+        indexExprs.push_back(getAffineDimExpr(i, rewriter.getContext()) -
+                             getAffineConstantExpr(1, context));
+      } else {
+        indexExprs.push_back(getAffineDimExpr(i, rewriter.getContext()));
+      }
+    }
+    auto cumsumResultMap = AffineMap::get(inputRank, 0, indexExprs, context);
+    SmallVector<AffineMap> indexingMaps = {
+        rewriter.getMultiDimIdentityMap(inputRank), cumsumResultMap};
+    SmallVector<StringRef> iteratorTypes(inputRank, "parallel");
+    SmallVector<Value> resultOutputDims;
+    for (auto i = 0; i < inputRank; i++)
+      resultOutputDims.push_back(getDimOp(rewriter, loc, input, i));
+    Value cumsumTensor =
+        createZeroInitTensor(rewriter, loc, resultOutputDims, elementType);
+    // llvm::errs()<<"before\n";
+    auto cumsumResult =
+        rewriter
+            .create<linalg::GenericOp>(
+                loc, cumsumTensor.getType(), input, cumsumTensor, indexingMaps,
+                iteratorTypes,
+                [&](OpBuilder &b, Location loc, ValueRange args) {
+                  Value currIndex = b.create<linalg::IndexOp>(loc, 0);
+                  Value result = b.create<arith::AddFOp>(loc, args[0], args[1]);
+                  Value indexEqualToZero = b.create<arith::CmpIOp>(
```

Comment:
We need to verify from a log but my understanding is that `currIndex` would start from 1 and never equal to 0 because in such case the indexing of `cumsumTensor` would be out of bound (currIndex - 1 < 0) and linalg generic op should guarantee the indexing is never out of bound according to https://mlir.llvm.org/docs/Dialects/Linalg/.  Could you verify that?

Reply:
I have verified it. It throws error ``linalg.generic' op unexpected result less than 0 at expression #0 in (d0) -> (d0 - 1)``